File: androidApp/src/main/java/com/picprogress/ds/components/card/AlbumCardContentSmall.kt
@file:OptIn(ExperimentalLayoutApi::class)

package com.picprogress.ds.components.card

import androidx.compose.foundation.layout.ExperimentalLayoutApi
import androidx.compose.foundation.layout.FlowRow
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.res.stringResource
import com.picprogress.R
import com.picprogress.model.album.Album
import com.picprogress.model.album.AlbumSummary
import com.picprogress.model.album.Frequency
import com.picprogress.model.album.isCurrent
import com.picprogress.util.getWeekNumber
import com.picprogress.util.name

@Composable
fun AlbumCardContentSmall(
    album: Album,
    summary: AlbumSummary
) {
    FlowRow {
        val currentFrame = summary.allFrames.first { it.isCurrent() }
        val text = when (album.frequency) {
            Frequency.DAILY -> currentFrame.start.dayOfWeek.name()
            Frequency.MONTHLY -> currentFrame.start.month.name()
            Frequency.WEEKLY -> {
                val weekNumber = currentFrame.start.getWeekNumber(album.createdAt.date)
                stringResource(id = R.string.album_card_week_label, weekNumber)
            }
        }

        FloatingTimeFrame(
            text = text,
            isCurrent = false,
            isCompleted = summary.completedFrames.contains(currentFrame),
            theme = album.theme,
            textAlignment = Alignment.TopStart
        )
    }
}

File: iosApp/iosApp/Modules/DS/Components/Card/AlbumCardContentSmall.swift
//
//  AlbumCardContentSmall.swift
//  PicProgress
//
//  Created by Jeferson on 01/02/24.
//  Copyright © 2024 orgName. All rights reserved.
//

import SwiftUI
import shared

struct AlbumCardContentSmall: View {
    
    let album: Album
    let summary: AlbumSummary
    
    var body: some View {
        
        let currentFrame = summary.allFrames.first(where: { $0.isCurrent() })
        let frameText: String? = currentFrame.flatMap { frame in
            switch album.frequency {
            case .daily:
                return frame.start.dayOfWeek.name(style: .full)
            case .monthly:
                return frame.start.month.name(style: .full)
            default:
                let weekNumber = frame.start.getWeekNumber(referenceDate: album.createdAt.date)
                return String(format: NSLocalizedString("Week", comment: ""), weekNumber)
            }
        }
        
        if let frameText = frameText, let currentFrame = currentFrame {
            FloatingTimeFrame(
                text: frameText,
                isCurrent: false,
                isCompleted: summary.completedFrames.contains(currentFrame),
                theme: album.theme,
                textAlignment: .topLeading,
                animate: false
            )
        } else {
            EmptyView()
        }
    }
}

